### PR TITLE
enhance performance for SuperRare nft events

### PR DESCRIPTION
**Pull request by jeff-dude**
using recently merged #1648 as reference, update query to be more performant on spark. this came up when monitoring in production.

### CODE REVIEW

**Comment by jeff-dude** [comment]
@catherine-2 please see this PR for attempt at enhanced performance for SuperRare events. in prod, it's taking a bit more time than it likely should each run.

i've written a query to hopefully expedite your review:
https://dune.com/queries/1369408

in there i have the ```original``` CTE which is your model, then ```jeff_test``` CTE which is the rewrite. i wrote an except between ```original``` and ```jeff_test``` CTE which returned no rows (should be good sign)

the main approach changes:
- inner joins to base tables where possible, i.e. ```ethereum.transactions```
- avoid using ```IN``` statement on query returned results, especially string fields like ```hash``` or such
- read from large tables as little as possible

spark seems to like this approach more. hope this makes sense and i didn't break anything!

**Comment by jeff-dude** [comment]
fyi @soispoke

**Comment by jeff-dude** [comment]
> @catherine-2 please see this PR for attempt at enhanced performance for SuperRare events. in prod, it's taking a bit more time than it likely should each run.
> 
> i've written a query to hopefully expedite your review: https://dune.com/queries/1369408
> 
> in there i have the `original` CTE which is your model, then `jeff_test` CTE which is the rewrite. i wrote an except between `original` and `jeff_test` CTE which returned no rows (should be good sign)
> 
> the main approach changes:
> 
> * inner joins to base tables where possible, i.e. `ethereum.transactions`
> * avoid using `IN` statement on query returned results, especially string fields like `hash` or such
> * read from large tables as little as possible
> 
> spark seems to like this approach more. hope this makes sense and i didn't break anything!

although the gh action failed on ```unique_trade_id``` so something must be off 🤔 likely if i open the filter up to more than 1 week, we'd see the issues (gh action was full history)

File: models/superrare/superrare_ethereum_events.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{ config(
     alias = 'events',
-    partition_by = ['block_time'],
+    partition_by = ['block_date'],
```
Comment (jeff-dude) [inline models/superrare/superrare_ethereum_events.sql:3]: another important minor tweak to call out: the partition field on date vs. time

if it's time, then we have a massive amount of directories in data lake with tiny files vs. less directories with larger files. spark prefers the latter. this also matches our base table design on tables such as ```ethereum.transactions```

**Comment by catherine-2** [comment]
Thank you so much @jeff-dude ! This is super helpful and definitely much cleaner than my previous version (and thank you for cleaning up some duplicate comments throughout too!). Will make a note to myself to make sure I'm better about using inner joins in spells. 

I went through all of the changes + compared the output and agree I think everything looks good! Though I'm also not sure what is causing the error with unique_trade_id.

**Comment by jeff-dude** [comment]
@catherine-2 i made another round of tweaks & the job runs fine with test passing for no dupes on ```unique_trade_id``` -- do you want to give a last glance before potentially merging and refreshing prod?

**Comment by catherine-2** [comment]
@jeff-dude I think this looks great! 

Thank you again for your help improving this spell -- really helpful to have a second set of eyes and gives me ideas to make my future queries better :)

**Review by soispoke — APPROVED** [review]
lgtm!

**Pull request merged**